src/service/resources-impl.js

```diff
@@ -1335,8 +1331,7 @@ export class Resources {
             this.requestsChangeSize_.push(request);
           }
           continue;
-        } else if (iniBox.bottom >= docBottomOffset ||
-                      box.bottom >= docBottomOffset) {
+        } else if (this.elementNearBottom_(box, iniBox)) {
```

Comment:
Nit: `iniBox` is only needed in this case. We could defer creating that struct until we call this method.

Reply:
Good idea.